application/classes/Controller/Api/Exports/External/Cli.php

```diff
@@ -0,0 +1,100 @@
+<?php defined('SYSPATH') OR die('No direct access allowed.');
+
+/**
+ * Ushahidi API Export Execute CLI Controller
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application\Controllers
+ * @copyright  2018 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\BufferedOutput;
+
+class Controller_Api_Exports_External_Cli extends Ushahidi_Rest {
+
+	protected function _scope()
+	{
+		return 'export_jobs';
+	}
+
+	public function checkApiKey($data)
+	{
+
+		if (isset($data['api_key'])) {
+			// Get api key and compare
+			return service('repository.apikey')->apiKeyExists($data['api_key']);
+		}
+
+		return false;
+	}
+
+	public function checkSignature($data)
+	{
+		$signature = $this->request->headers('X-Ushahidi-Signature');
+
+		if ($signature) {
+			//Validate signature
+			$shared_secret = getenv('PLATFORM_SHARED_SECRET');
+			$signer = new Signer($shared_secret);
+			$fullURL = URL::site(Request::detect_uri(), TRUE) . URL::query();
+
+			return $signer->validate($signature, $fullURL, $data);
+		}
+		return false;
+	}
+
+	// public function before()
+	// {
+	// 	parent::before();
+
+	// 	$post = $this->_request_payload;
+
+	// 	if (!$this->checkApiKey($post) || !$this->checkSignature($post))
+	// 	{
+	// 		throw HTTP_Exception::factory(403, 'Forbidden');
+	// 	}
+	// }
+
+	protected function _is_auth_required()
+	{
+		return false;
+	}
+
+	public function action_get_index($command = 'export')
+	{
+
+		// Get Symfony console app
+		$app = service('app.console');
+		$command = $app->get('exporter');
+		$limit = 0;
+		$offset = 0;
+
+		$job_id = $this->request->param('id');
+
+		// Construct console command input
+		$input = new ArrayInput(array(
+			'action' => 'export',
+			'--limit' => $limit,
+			'--offset' => $offset,
+			'--job' => $job_id,
+		 ), $command->getDefinition());
+		 
+
+		// Create Output Buffer
+		$output = new BufferedOutput();
+
+		
+		// Run the command
+		$command->run($input, $output);
+
+		//Retrieve results
+		$export_results = json_decode($output->fetch());
+
+		$this->_response_payload = [
```

Comment:
Sorry, this is a slight misnomer I'll rename, what is actually returned is the url of the file as a json formatted string, this is because the CLI tool prints to the command line a string of serialised json which we then read and then re-encode as json.

Fixed by changing name and improving with a comment.

Reply:
thanks ! that makes sense